Add ability to ignore unknown fields in kafka messages

podgaietska opened the pull request:
### Summary
This change introduces a new connector configuration option to control how unknown fields in Kafka records are handled when writing to BigQuery.

### Details
- Added a config flag ignoreUnknownFields (default: false).
- When enabled, fields not present in the BigQuery table schema are silently ignored instead of causing the record to be rejected.
- Applied to both ingestion modes: insertAll and StorageWriteApi

Claudenw commented:
@RyanSkraba What is your opinion of this PR.  I  think that it is useful and looks like it will work correctly, but... It makes changes to existing public APIs, though perhaps they should not be considered public APIs as they are intended to be used internally.  What are your thoughts.

@SamoylovMD You said you have a fork of this project that you are trying to resolve by adding the functionality to this repository.  A tack that I appreciated.  My question to you is how much damage would the changes here be for you?

@podgaietska Could you, rather than change the constructors, deprecated the current constructors and add a constructor that accepts the BigQuerySinkConfig?  This way, in the future when new options are added we do not have to change all the methods again.

RyanSkraba commented on kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/StorageWriteApiBigQuerySinkConnectorIT.java:
```diff
@@ -307,6 +307,97 @@ public void testAvroWithSchemaUpdate() throws InterruptedException {
+            TASKS_MAX,
+            "Tasks should have failed when writing record with fields not present"
+                    + "in target table." );
+  } @Test public void testRecordWithUnknownFieldWhenIgnoreEnabled() throws InterruptedException {
```
Some odd formatting hiere!  Can you fix this?

podgaietska replied:
Good catch. Fixed. Thanks!

Claudenw reviewed (changes requested):
The deprecated tags need to come at the end of the javadoc.   if you run `mvn compile` you will see all the lines that need to be changed.

Note:  
When I wrote about passing the config object I was intending that calls to 
`config.getBoolean(BigQuerySinkConfig.IGNORE_UNKNOWN_FIELDS_CONFIG);` would be contained within the config class itself as something like:

```
public boolean ignoreUnknownFields() {
  return getBoolean(BigQuerySinkConfig.IGNORE_UNKNOWN_FIELDS_CONFIG);
}
```
This helps enforce the concept of information hiding, where the calling code does not need to know how we figure out if unknown fields are ignored, just whether or not we do.

calling code then looks like 
```
this.ignoreUnknownFields = config.ignoreUnknownFields();
```

However, since the rest of the code base uses the same style as your solution, I will approve this once the tests pass.

Also, some developers and projects will state that `ignoreUnknownFields()` should be 'isIgnoreUnknownFields()`.  I can go either way on this argument, sometimes `is` seems to make more sense sometimes not.

podgaietska commented on kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java:
```diff
@@ -931,6 +938,12 @@ public boolean visible(String s, Map<String, Object> map) {
+            IGNORE_UNKNOWN_FIELDS_TYPE,
+            IGNORE_UNKNOWN_FIELDS_DEFAULT,
+            IGNORE_UNKNOWN_FIELDS_IMPORTANCE,
+            IGNORE_UNKNOWN_FIELDS_DOC
```
Changed. Thanks!

Claudenw commented:
If you look at the output of the run you will see that some of the integration tests failed because there was not a value of something.  I suspect the something was an item you added.  We need to ensure that the default configuration will work as before without the fields you added.

podgaietska commented:
> If you look at the output of the run you will see that some of the integration tests failed because there was not a value of something. I suspect the something was an item you added. We need to ensure that the default configuration will work as before without the fields you added.

@Claudenw hi! The issue was caused by a change I made in PR #69 that was just merged, where I updated the `initialiseJsonConverters` method to take a boolean argument. The two new tests I added in this PR  (`testRecordWithUnknownFieldFails` and `testRecordWithUnknownFieldWhenIgnoreEnabled`) were still calling the old no-arg version, which led to the compilation errors.

I’ve pushed a fix. Please attempt to rerun. Thanks!
